### PR TITLE
Add nginx status and logging for builder API

### PR DESCRIPTION
**Pull request by chefsalim**
Update the nginx configuration for builder API to add an internal status endpoint, along with custom nginx call logging.

Signed-off-by: Salim Alam <salam@chef.io>

tenor-209923620

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by nellshamrell** [comment]
👍

File: components/builder-api-proxy/config/nginx.conf [hunk 1]
```diff
@@ -70,11 +74,21 @@ http {
       rewrite ^(.*)$ https://$host$1 permanent;
     }
 
+    access_log {{pkg.svc_path}}/logs/host.access.log nginx;
```
Comment (reset) [inline components/builder-api-proxy/config/nginx.conf]: Will this rotate appropriately? What we really want to do is ensure that logs from journald get into a log aggregator like Sumo Logic (unless data dog also does that)

Comment (reset) [reply components/builder-api-proxy/config/nginx.conf]: Or maybe since it's the access log we really don't care for it in an aggregator?

Comment (chefsalim) [reply components/builder-api-proxy/config/nginx.conf]: The Datadog agent will be able to parse it and we'll get the stats that way. The rotation is something I will figure out shortly - this is likely the first of a few tweaks on this.

Comment (fnichol) [reply components/builder-api-proxy/config/nginx.conf]: Yeah, we'll want to be careful about rotation or max file size. We've already learned that high traffic directly affects syslog file sizes which isn't surprising.

Comment (fnichol) [reply components/builder-api-proxy/config/nginx.conf]: Yuck, looks like nginx doesn't handle this itself: https://www.nginx.com/resources/wiki/start/topics/examples/logrotation/

Comment (chefsalim) [reply components/builder-api-proxy/config/nginx.conf]: Yeah, I'm leaning toward using logrotate - it looks fairly straightforward to wire up.

Comment (reset) [reply components/builder-api-proxy/config/nginx.conf]: The other option is to rely solely on your aggregator. I did this on my last large project and it worked swimmingly

Comment (chefsalim) [reply components/builder-api-proxy/config/nginx.conf]: Yeah the main reason atm to log to file with nginx is to get the response times

**Comment by fnichol** [comment]
Would it be possible to also include #3063 in this PR?

**Pull request merged**